### PR TITLE
add support for PCI vendor-specific capability

### PR DESCRIPTION
**Pull request by tmakatos**
I've included a couple of minor, trivial patches in this PR.

### CODE REVIEW

File: lib/libvfio-user.c [hunk 1]
```diff
@@ -284,6 +284,10 @@ pci_config_space_size(vfu_ctx_t *vfu_ctx)
     return region_size(vfu_ctx, VFU_PCI_DEV_CFG_REGION_IDX);
 }
 
+/*
+ * Accesses the non-standart part (offset >= 0x40) of the PCI configuration
```
Comment (jlevon) [inline lib/libvfio-user.c:288]: nit, "standard"

File: samples/lspci.c [hunk 2]
```diff
@@ -44,9 +44,10 @@ int main(void)
     const int bytes_per_line = 0x10;
     vfu_pci_hdr_id_t id = { 0 };
     vfu_pci_hdr_ss_t ss = { 0 };
-    vfu_pci_hdr_cc_t cc = { { 0 } };
+    vfu_pci_hdr_cc_t cc = { 0 };
```
Comment (jlevon) [inline samples/lspci.c:47]: Looks like this is reverting the GCC4 build fix?

File: test/unit-tests.c [hunk 3]
```diff
@@ -356,6 +358,98 @@ test_attach_ctx(void **state __attribute__((unused)))
...
+    assert_int_equal(8,
+                     config_space.raw[PCI_STD_HEADER_SIZEOF + PCI_PM_SIZEOF + PCI_CAP_LIST_NEXT + 1]);
+    assert_int_equal(PCI_CAP_ID_VNDR,
+                     config_space.raw[PCI_STD_HEADER_SIZEOF + PCI_PM_SIZEOF + vsc[0]->vsc.size]);
+    assert_int_equal(0,
+                     config_space.raw[PCI_STD_HEADER_SIZEOF + PCI_PM_SIZEOF + vsc[0]->vsc.size + PCI_CAP_LIST_NEXT]);
+    assert_int_equal(vsc[1]->vsc.size,
+                     config_space.raw[PCI_STD_HEADER_SIZEOF + PCI_PM_SIZEOF + vsc[0]->vsc.size + PCI_CAP_LIST_NEXT + 1]);
+
+    /*
```
Comment (jlevon) [inline test/unit-tests.c:429]: I'm not following this FIXME, didn't we just do that?

**Review by jlevon — CHANGES_REQUESTED** [review]

File: samples/lspci.c (hunk 2 above)
Comment (tmakatos) [reply samples/lspci.c:47]: Yes, sorry, will fix.

File: lib/libvfio-user.c (hunk 1 above)
Comment (tmakatos) [reply lib/libvfio-user.c:288]: ok

File: test/unit-tests.c (hunk 3 above)
Comment (tmakatos) [reply test/unit-tests.c:429]: right

**Comment by tmakatos** [comment]
@jlevon addressed your comments, please take another look

**Review by jlevon — APPROVED** [review]
Thanks, but reminder, please don't force-push - I lost the diffs since last review as a result.

**Comment by tmakatos** [comment]
> Thanks, but reminder, please don't force-push - I lost the diffs since last review as a result.

Sorry for that, I think I've misunderstood in which cases github retains the previous diff.

**Pull request merged**